templates/config/defaults.json

```diff
@@ -19,9 +19,16 @@
         "@id": "urn:solid-server:template:variable:baseUrl"
       }
     },
-
     {
       "@id": "urn:solid-server:template:ResourceStore",
+      "@type": "RepresentationConvertingStore",
+      "metadataStrategy":{ "@id": "urn:solid-server:default:MetadataStrategy" },
+      "options_inConverter": { "@id": "urn:solid-server:default:RepresentationConverter" },
+      "options_outConverter": { "@id": "urn:solid-server:default:UiEnabledConverter" },
+      "source": { "@id": "urn:solid-server:template:ResourceStore_Backend" }
```

Comment:
Why did this need to be added? Was the converting store in the base config not sufficient? If not a comment should be added to explain.

Reply:
This template config just used as a `urn:solid-server:template:ResourceStore` just the `DataAccessorBasedStore`. 
This formed an error for creating metadata during startup for the `DynamicPods`. 

With the full ResourceStore in the https://github.com/CommunitySolidServer/CommunitySolidServer/blob/main/config/storage/middleware/default.json everything is okay as the `RepresentationConvertingStore` comes to the rescue to convert metadata to `INTERNAL/QUADS`, but in templates it just didn't convert hence the errors and the reason I added it to the config.

Reply:
Instead of doing this, could you try doing the following change?

Change the line at https://github.com/CommunitySolidServer/CommunitySolidServer/blob/9dcba1a2881b5568a57d4a9fa287917ea32193c7/src/pods/ConfigPodManager.ts#L46
to use the standard resource store (so `urn:solid-server:default:ResourceStore`, which `ConfigPodManager` should take as input parameter then). That way the generated resources will go through the representation converter.

Otherwise the templated stores will have an extra converting store in their chain which is only relevant during generation.

Reply:
Addendum: the line `await this.routingStorage.set(identifier.path, store);` will probably have to be moved to be above the `addGeneratedResources` call, otherwise the routing store won't be able to find the pod store yet.